astroquery/mast/auth.py

```diff
@@ -82,10 +81,8 @@ def login(self, token=None, store_token=False, reenter_token=False):
         if not info["anon"]:
             log.info("MAST API token accepted, welcome {}".format(info["attrib"].get("display_name")))
         else:
-            warn_msg = ("MAST API token invalid!\n"
-                        "To make create a new API token visit to following link: " +
-                        self.AUTH_URL)
-            warnings.warn(warn_msg, AuthenticationWarning)
+            raise LoginError("MAST API token invalid!\n To make create a new API token"
+                             "visit to following link: " + self.AUTH_URL)
 
         return not info["anon"]
```

Comment:
Since we are now raising an error it is maybe not necessary to return success/failure. But removing this would change the function behavior and potential break client checks. Thoughts @bsipocz?

Reply:
Just thinking...we could leave it as a warning for now but also give a deprecation period with a deprecation message that notifies clients of this incoming change. That is if raising an error is really the intent here. That way clients would have time to change and adopt the exception flow. Thoughts @ceb8  @bsipocz ?

Reply:
I would say this one should change as `self._authenticated` is being set to be this return. (However the whole business with `self._authenticated` is a bit of  a state, e.g.  it's only set to `False` after logout, but not when it's not token authenticated, so a bit of cleanup would be indeed nice. But I think that goes beyond this PR, so I'll rather open a separate issue for it.